internal/webhook/proxmoxmachine_webhook.go

```diff
@@ -131,3 +144,14 @@ func validateNetworkDeviceMTU(device *infrav1.NetworkDevice) error {
 
 	return nil
 }
+
+func validateTags(tags []string) error {
+	re := regexp.MustCompile(`^[a-zA-Z0-9-_]*$`)
```

Comment:
Should be `+` not `*` to prevent empty string. Also, have you tried `+kubebuilder:validation:Pattern`?

Reply:
Hmmm, 

Let me check